app/project/author/controllers.py

```diff
@@ -0,0 +1,72 @@
+from flask_restful import Resource
+
+from project.db import db, Author, Book
+from project.utils.auth import require_admin
+from .parsers import AuthorAddParser, AuthorParser
+
+class List(Resource):
+
+    def get(self, user_data):
+        return [a.json for a in Author.query.all()]
+
+class Add(Resource):
+
+    @require_admin
+    def post(self, user_data):
+        parser = AuthorAddParser(bundle_errors=True)
+        args = parser.parse_args()
+
+        author = Author()
+        author.name = args['name']
+        author.surname = args['surname']
+        author.www = args['www']
+        author.slug = "{0}-{1}".format(args['name'].lower(), args['surname'].lower())
```

Comment:
there is a better tool in sqlalchemy to handle thinks like auto-generated slugs. I think it should work:
```
from sqlalchemy.event import listen

class Author(db.Model):
    ....
    def generate_slug(self):
        self.slug = '{}-{}'.format(self.name, self.surname)

listen(User, 'before_insert', generate_license)
```